src/webgpu/api/operation/buffers/map.spec.ts

```diff
@@ -367,12 +367,19 @@ g.test('mappedAtCreation,mapState')
   .desc('Test that exposed map state of buffer created with mappedAtCreation has expected values.')
   .params(u =>
     u
-      .combine('validationError', [false, true])
+      .combine('usageType', ['invalid', 'MAP_READ', 'MAP_WRITE'])
       .combine('afterUnmap', [false, true])
       .combine('afterDestroy', [false, true])
   )
   .fn(t => {
-    const { validationError, afterUnmap, afterDestroy } = t.params;
+    const { usageType, afterUnmap, afterDestroy } = t.params;
+    const usage =
+      usageType === 'MAP_READ'
+        ? GPUBufferUsage.COPY_DST | GPUBufferUsage.MAP_READ
+        : usageType === 'MAP_WRITE'
+        ? GPUBufferUsage.COPY_SRC | GPUBufferUsage.MAP_WRITE
+        : 0;
```

Comment:
Another option might be

```typescript
const usages = {
  invalid: 0,
  MAP_READ: GPUBufferUsage.COPY_DST | GPUBufferUsage.MAP_READ,
  MAP_WRITE: GPUBufferUsage.COPY_SRC | GPUBufferUsage.MAP_WRITE,
};
...
const usage = usages[usageType];
```

Reply:
No strong feeling, I think both styles are fine but I usually use the object-based one. Sometimes inline like:
`const usage = { invalid: 0, /* ... */ }[usageType];`

Some other tests use things like:
```
.combine('usage', [0, GPUConst.BufferUsage.COPY_DST | GPUConst.BufferUsage.MAP_READ, GPUConst.BufferUsage.COPY_SRC | GPUConst.BufferUsage.MAP_WRITE])
```

but that also makes the case strings kind of unreadable so I think what you have now is better.

Reply:
Thanks for the comment. Then I want to go as is.